drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -309,8 +265,48 @@ impl ServerImpl {
         res
     }
 
-    // Fetch as many bytes as we can and parse the header.
+    fn do_send_request(&mut self, size: usize) -> Result<(), SprotError> {
+        ringbuf_entry!(Trace::SendRecv(size));
+        let buf = &self.tx_buf.as_slice()[..size];
+
+        // In order to improve reliability, start by sending only the
+        // first ROT_FIFO_SIZE bytes and then delaying a short time.
+        // If the RoT is ready, those first bytes will always fit
+        // in the RoT receive FIFO. Eventually, the RoT FW will respond
+        // to the interrupt and enter a tight loop to receive.
+        // The short delay should cover most of the lag in RoT interrupt
+        // handling.
+        let part1 = if let Some(part1) = buf.get(0..ROT_FIFO_SIZE.min(size)) {
+            part1
+        } else {
+            return Err(SprotError::BadMessageLength);
+        };
+        let part2 = buf.get(part1.len()..).unwrap_lite(); // empty or not
+        ringbuf_entry!(Trace::TxPart1(part1.len()));
+        ringbuf_entry!(Trace::TxPart2(part2.len()));
+        ringbuf_entry!(Trace::CSnAssert);
+        self.spi
```

Comment:
We can simplify this to remove the 3 `map_err` calls by implementing `From<SpiError> for SprotError`

Reply:
done